deep_qa/models/multiple_choice_qa/tuple_inference.py

```diff
@@ -175,3 +177,99 @@ def _build_model(self):
         final_output = MaskedSoftmax()(options_probabilities)
 
         return DeepQaModel(input=[question_input, background_input], output=[final_output])
+
+    def debug_display_instance_tuple_match_scores(self, answer_tuples, background_tuples, tuple_matcher_output,
+                                                  instance, num_to_display):
+        num_background = min(self.num_background_tuples, len(instance.background_tuples))
+        result = ""
+        for i, answer_tuple in enumerate(answer_tuples):
+            answer_tuple_string = "\n\t".join(textwrap.wrap(answer_tuple.display_string(), 150))
+            result += "Question (repeated): %s\n" % instance.question_text
+            result += "Answer_tuple_{0} : \n\t{1}\n\n".format(i, answer_tuple_string)
+            result += "Top {0} (out of {1}) highest scoring background tuples:\n\n".format(num_to_display,
+                                                                                           num_background)
+            storage = []
+            for j, background_tuple in enumerate(background_tuples):
+                tuple_match_score = tuple_matcher_output[i, j]
+                storage.append((tuple_match_score, j, background_tuple))
+            # Sort descending by tuple match score
+            sorted_by_score = sorted(storage, key=lambda tup: tup[0], reverse=True)[:num_to_display]
+            for scored in sorted_by_score:
+                background_tuple_index = scored[1]
+                background_tuple_string = scored[2].display_string()
+                wrapped_tuple = "\n\t".join(textwrap.wrap(background_tuple_string, 150))
```

Comment:
You could consider making this a parameter to the class, instead of a hard-coded number here.  Same with `num_to_display = 5` below.